### DISCUSSION TITLE
Documentation error on vercel.json "missing" usage

### DISCUSSION THREAD

**Discussion by ubergarm** [vercel/community#2920] [Bug Report]
### Page to Investigate

https://vercel.com/docs/concepts/projects/project-configuration

### Steps to Reproduce

The documentation `header "has" definition` at https://vercel.com/docs/concepts/projects/project-configuration#header-has-object-definition shows incorrect usage regarding the use of a "missing" block. The documentation incorrectly suggest in multiple places that "missing" is a keyword of a "has" block which is incorrect as of Vercel CLI 30.2.2

Using the provided examples of "missing" as a keyword of a "has" block gives the error 
```
"Error: Invalid vercel.json - `redirects[0].has[0]` should NOT have additional property `key`. Please remove it."
```

You can see from the source code that "missing" is not a keyword of "has" but it actually uses the same schema as a "has" block: https://github.com/vercel/vercel/blob/main/packages/routing-utils/src/schemas.ts#L167-L168

The `vercel.json` example shown at https://vercel.com/docs/concepts/projects/project-configuration#rewrites is invalid JSON (the last block in question is missing commas and improperly indented), and it does not work.

Below is an example working `vercel.json` that shows the proper usage of the "missing" block which has the exact same schema as a "has" block. This example handles rewriting parameters for a SPA and also applies basic auth headers to the content. Replace "base64:encodethis" with your desired values e.g.

```bash
$ echo -n "aladdin:opensesame" | base64
YWxhZGRpbjpvcGVuc2VzYW1l
```

`vercel.json`:
```json
{
  "routes": [
    {
      "src": "/[^.]+",
      "dest": "/",
      "status": 200,
      "has": [
        {
          "type": "header",
          "key": "authorization",
          "value": "Basic base64:encodethis"
        }
      ]
    },
    {
      "src": "/.*",
      "status": 401,
      "headers": { "www-authenticate": "Basic realm=\"staging\"" },
      "missing": [
        {
          "type": "header",
          "key": "authorization",
          "value": "Basic base64:encodethis"
        }
      ]
    }
  ]
}
```

**Comment by smaeda-ks** [comment]
@ubergarm Good catch. Thanks for flagging this! We will update the docs accordingly

**Comment by timeyoutakeit** [comment]
Hi @ubergarm, apologies for this oversight but thank you for bringing it to our attention! We've pushed a fix to the docs with updated examples and references: https://vercel.com/docs/concepts/projects/project-configuration#rewrites

**Reply by ubergarm** [reply]
Just did a quick review - lgtm! Thanks for the quick turn-around! I got my desired setup working great `routes` + `missing` similar to how you describe in the docs!
Cheers!
